src/tests/unit/userService/comparePassword.test.ts

```diff
@@ -0,0 +1,22 @@
+import UserService from '../../../services/userService';
+import { mockPassword } from '../../mocks/mockUser';
+import { setupTests } from '../../setupTests';
+import bcrypt from 'bcrypt';
+
+const userService = new UserService();
+
+setupTests('comparePassword', () => {
+  test('returns true if the password matches hashed password', async () => {
+    const hashedPassword = await bcrypt.hash(mockPassword, 10);
+    expect(
+      await userService.comparePassword(mockPassword, hashedPassword)
+    ).toBe(true);
+  });
+
+  test("returns false if the password doesn't match hashed password", async () => {
+    const hashedPassword = await bcrypt.hash(mockPassword, 10);
+    expect(await userService.comparePassword('wrong', hashedPassword)).toBe(
+      false
+    );
+  });
```

Comment:
I'd add test to make sure that the hash created by `userService.createUser` is comparable with the password that has been put as input

something like
1. define create user payload
2. pass this payload to `userService.createUser`
3. get the hash from the created user`
4. compare the password from create user payload with the hash from the created user
5. assert that comparison returned truth value